runtime/launcher.m

```diff
@@ -429,29 +430,42 @@
 	mono_version = mono_get_runtime_build_info ();
 	if (!check_mono_version (mono_version, [minVersion UTF8String]))
 		exit_with_message ([[NSString stringWithFormat:@"This application requires the Mono framework version %@ or newer.", minVersion] UTF8String], data->basename, true);
-
-	// 6) Find the executable. The name is: [...]
-	NSString *exeName = NULL;
-	NSString *exePath;
-	if (plist != NULL)
-		exeName = (NSString *) [plist objectForKey:@"MonoBundleExecutable"];
-	else
-		fprintf (stderr, PRODUCT ": Could not find Info.plist in the bundle.\n");
-
-	if (exeName == NULL)
-		exeName = [[NSString stringWithUTF8String: data->basename] stringByAppendingString: @".exe"];
-	
-	if (mkbundle) {
-		exePath = exeName;
-	} else {
-		exePath = [[[NSString stringWithUTF8String: xamarin_get_bundle_path ()] stringByAppendingString: @"/"] stringByAppendingString: exeName];
-
-		if (!xamarin_file_exists ([exePath UTF8String]))
-			exit_with_message ([[NSString stringWithFormat:@"Could not find the executable '%@'\n\nFull path: %@", exeName, exePath] UTF8String], data->basename, false);
-	}
-
-	exe_path = strdup ([exePath UTF8String]);
-
+    // 6) Find the executable. The name is: [...]
+    if (!is_extension) {
+        NSString *exeName = NULL;
+        NSString *exePath;
+        if (plist != NULL)
+            exeName = (NSString *) [plist objectForKey:@"MonoBundleExecutable"];
+        else
+            fprintf (stderr, PRODUCT ": Could not find Info.plist in the bundle.\n");
+        
+        if (exeName == NULL)
+            exeName = [[NSString stringWithUTF8String: data->basename] stringByAppendingString: @".exe"];
+        
+        exePath = [[[NSString stringWithUTF8String: xamarin_get_bundle_path ()] stringByAppendingString: @"/"] stringByAppendingString: exeName];
```

Comment:
Looks like it might have gotten wiped out into the rebase onto latest (at the time) master.  I'll get it fixed